docs/basic/getting-started/hooks.md

```diff
@@ -87,18 +89,43 @@ When using `useEffect`, take care not to return anything other than a function o
 ```ts
 function DelayedEffect(props: { timerMs: number }) {
   const { timerMs } = props;
-  // bad! setTimeout implicitly returns a number because the arrow function body isn't wrapped in curly braces
+
   useEffect(
     () =>
       setTimeout(() => {
         /* do stuff */
       }, timerMs),
     [timerMs]
   );
+  // bad example! setTimeout implicitly returns a number
+  // because the arrow function body isn't wrapped in curly braces
   return null;
 }
 ```
 
+<details>
+<summary>
+Solution to the above example
+</summary>
+
+```ts
+function DelayedEffect(props: { timerMs: number }) {
+  const { timerMs } = props;
+
+  useEffect(
+    () =>
+      void setTimeout(() => {
+        /* do stuff */
+      }, timerMs),
+    [timerMs]
+  );
+  // better; use the void keyword to make sure you return undefined
+  return null;
+}
+```
+
+</details>
+
```

Comment:
I would use a prop `{}` block instead. The `void` keyword is quite esoteric.

Reply:
```suggestion
<details>
<summary>
Solution to the above example
</summary>

```ts
function DelayedEffect(props: { timerMs: number }) {
  const { timerMs } = props;

  useEffect(
    () => {
      setTimeout(() => {
        /* do stuff */
      }, timerMs)
    },
    [timerMs]
  );
  // better; use the void keyword to make sure you return undefined
  return null;
}
```

</details>

```